simulator/modules/core/perf_sim.h

```diff
@@ -55,6 +55,13 @@ class PerfSim : public CycleAccurateSimulator
     void write_cause_register( uint64 value) { write_register( Register::cause(), value); }
     void write_csr_register( std::string_view name, uint64 value) final { write_register( Register::from_csr_name( name), value); }
 
+    auto get_mispredict_rate( int total, float64 piece) const
```

Comment:
This method may be moved out of class and reside in perf_sim.cpp. It would improve compilation speed a little.